src/compiler.js

```diff
@@ -220,7 +220,18 @@ class Compiler {
         throw e
       }
 
+      // Look for `node_modules` in the same dir as `subgraph.yaml`.
       let libs = path.join(baseDir, 'node_modules')
+      if (!fs.existsSync(libs)) {
+        // It's not there, so look in the directories above. This, in practice, checks all ancestor
+        // directories but keeps the loop simple.
+        for (let i = 1; i <= 100; i++) {
+          let upwardPath = path.resolve(libs, ...Array(i).fill('..'), 'node_modules')
+          if (fs.existsSync(upwardPath)) {
+            libs = upwardPath
+          }
+        }
+      }
```

Comment:
Rather than the slightly obscure iteration over `i`, wouldn't something like this be better (and more obvious)?
```
const fs = require('fs')
const path = require('path')

let manifestFile = '/usr/local/share/subgraph.yaml'

let libs
for (
  let dir = baseDir;
  libs === undefined && dir !== undefined;
  dir = path.dirname(dir) === dir ? undefined : path.dirname(dir)
) {
  if (fs.existsSync(path.join(dir, 'local'))) {
    libs = dir
  }
}
```
You can play with it here: https://repl.it/repls/MintyStripedLinuxpc